cluster/manager/manager.go

```diff
@@ -622,13 +654,19 @@ func (c *ClusterManager) initNodeInCluster(
 	for e := c.listeners.Front(); e != nil; e = e.Next() {
 		finalizeCb, err := e.Value.(cluster.ClusterListener).Init(self, clusterInfo)
 		if err != nil {
-			if self.Status != api.Status_STATUS_MAINTENANCE {
-				self.Status = api.Status_STATUS_ERROR
+			initErr := err
```

Comment:
this does not look correct - if init fails for one listener, with no quorum member error, and we wait and quorum is established, we would continue initing other listeners - so some listeners init failed and for others it was successful ? 
we can simply return the error and cleanup. let listener do the waiting.

Reply:
Agreed. The assumption here was init has succeeded but the listener asked to wait. 

The problem with doing a cleanup is we will be cleaning up the Init done by this listener as well. Also currently if an error is returned from this function then  for the next retry the cluster manager has to go through its whole state machine again if it wants to proceed.

Instead can we change the listener's Init API itself to look something like this
func Init(api.Node, api.ClusterInfo) (waitForQuorum bool, waitForQuorumMessage string, error)

In this way the API itself is clear and the listener can return a nil error but still ask the cluster manager to wait for quorum.